engine/chain/internal/combined_txs_getter.go

```diff
@@ -61,11 +61,20 @@ func (ctg *CombinedTxsGetter) GetTransactions(ctx context.Context, ids iter.Seq[
 	return transactions, nil
 }
 
-func containsNilGetter(getters []chainmodels.TransactionsGetter) bool {
-	for _, getter := range getters {
-		if getter == nil {
-			return true
-		}
-	}
-	return false
+type emptyGetter struct{}
+
+func (ctg *emptyGetter) GetTransactions(_ context.Context, _ iter.Seq[string]) ([]*sdk.Transaction, error) {
+	return nil, nil
+}
+
+func filterNilGetters(txsGetters ...chainmodels.TransactionsGetter) []chainmodels.TransactionsGetter {
+	return slices.Collect(
+		func(yield func(chainmodels.TransactionsGetter) bool) {
+			for _, v := range txsGetters {
+				if v != nil && !yield(v) {
+					return
+				}
+			}
+		},
```

Comment:
I'm not sure if this is much cleaner then:
```suggestion
	result := make([]chainmodels.TransactionsGetter, 0)
	for _, getter := range txsGetters {
		if getter == nil {
			result = append(result, getter)
		}
	}
	return result
```

it would be if there would be some function `slices.Filter(txsGetters, func(i, item) bool { .... })` but I cannot find such thing 😞

Reply:
I've changed it to 
```golang
return slices.DeleteFunc(txsGetters, func(getter chainmodels.TransactionsGetter) bool {
		return getter == nil
	})
```